src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -153,6 +163,42 @@ int32_t AppleCryptoNative_SslSetTargetName(SSLContextRef sslContext,
     return *pOSStatus == noErr;
 }
 
+DLLEXPORT int32_t AppleCryptoNative_SslCtxSetAlpnProtos(SSLContextRef sslContext,
+                                                     const uint8_t* protocols,
+                                                     int32_t len,
+                                                     int32_t* pOSStatus)
+{
+    if (sslContext == NULL || protocols == NULL || pOSStatus == NULL || pOSStatus == NULL)
+        return -1;
+
+    *pOSStatus = SSLSetALPNData(sslContext, protocols, (size_t)len);
+
+    return *pOSStatus == noErr;
+}
+
+DLLEXPORT int32_t AppleCryptoNative_SslGetAlpnSelected(SSLContextRef sslContext, const uint8_t** protocol, uint32_t* len)
+{
+    size_t  alpnLen = 0;
+    const uint8_t * ext;
+    if (sslContext == NULL || protocol == NULL || len == NULL)
+        return -1;
+
+    ext = SSLGetALPNData(sslContext, &alpnLen);
+
+    if (ext == NULL ||  alpnLen < 1)
+    {
+        *len = 0;
+        *protocol = NULL;
+        return 0;
+    }
+
+    // Get protocol data from TLS extension
+    *len = (uint32_t)alpnLen - 1;
```

Comment:
Missing an overflow check.  `SizeTToInt32` would do it if the output was a signed integer.